sklearn/decomposition/dict_learning.py

```diff
@@ -309,19 +309,28 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
     return code
 
 
-def _update_dict(dictionary, Y, code, verbose=False, return_r2=False,
-                 random_state=None):
+def _proj_l2(v):
+    """Projects v unto l2 unit ball in-place.
+    """
+    vv = np.dot(v, v)
+    if vv > 1.:
+        v /= sqrt(vv)
+    return v
+
+
+def _update_dict(dictionary, B, A, verbose=False, return_r2=False,
+                 random_state=None, online=False):
     """Update the dense dictionary factor in place.
 
     Parameters
     ----------
     dictionary : array of shape (n_features, n_components)
         Value of the dictionary at the previous iteration.
 
-    Y : array of shape (n_features, n_samples)
+    B : array of shape (n_features, n_components)
         Data matrix.
 
-    code : array of shape (n_components, n_samples)
+    A : array of shape (n_components, n_components)
```

Comment:
Ya, indeed it looks like the old `_dict_update` code is incorrect in the online case, due to the extra `dictionary[:, k] = np.dot(R, A[k, :].T)` on line [357](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/decomposition/dict_learning.py#L357).